### PR TITLE
Bugfix 1715

### PR DESCRIPTION
**Pull request by bibo38**
To getting started, I fixed the Issue #1715 by replcaing some variable names.

### CODE REVIEW

File: src/Endianness.h [hunk 1]
```diff
@@ -11,10 +11,10 @@
 // Changes endianness
 inline UInt64 HostToNetwork8(const void * a_Value)
 {
-	unsigned long long __HostToNetwork8;
-	memcpy( &__HostToNetwork8, a_Value, sizeof( __HostToNetwork8));
-	__HostToNetwork8 = (( ( (unsigned long long)htonl((u_long)__HostToNetwork8)) << 32) + htonl(__HostToNetwork8 >> 32));
-	return __HostToNetwork8;
+	unsigned long long buf;
```
Comment (worktycho) [inline src/Endianness.h]: This should probably be changed to a UInt64.

**Comment by bibo38** [comment]
Shouldn't the Globals.h also be included when using the UInt64 typedefs?

**Comment by worktycho** [comment]
This project has a confusing convention of including Globals.h in the cpp files only. It has something to do with the MSVC build.

**Comment by madmaxoft** [comment]
The reason for Globals.h is Precompiled headers - MSVC uses that file to generate the precompiled header, which speeds up the compilation considerably. But then it needs to be the first #include in each and every .cpp file and it has to be the exact same path in each .cpp file.

I wish we had something similar for gcc, I have too little knowledge to set that up, but I believe it would speed up the compilation quite a lot, too.

**Comment by bibo38** [comment]
Maybe **[Cotire](https://github.com/sakra/cotire)** is the right way, which is mentioned in [this Stackoverflow question](http://stackoverflow.com/a/9838635): 

It should automaticly create Precompiled headers and a [Single Compilation Unit](http://en.wikipedia.org/wiki/Single_Compilation_Unit). Or you can use your own Precompiled Headers file.
All in all it is compatible with all common used compilers, so it should speed gcc and clang up.
I've never used it, but it seems to work really good.

**Pull request merged**